airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/manifest_component_transformer.py

```diff
@@ -56,7 +55,11 @@
     # AddFields
     "AddFields.fields": "AddedFieldDefinition",
     # CustomStreamSlicer
+    "CustomStreamSlicer.end_datetime": "MinMaxDatetime",
```

Comment:
noticed a couple of connectors where we need to auto-infer these values. some explicitly type them, but some connectors don't so we need this to auto insert it when it doesn't exist

Reply:
The guideline so far has just been to ensure backwards compatibility so we have been adding when we catch it during our custom component testing, but once we get this change through, the templates and docs should be updated and developers should always be specifying the `type` in every component. I think our original hypothesis of type inference might not have paid off because even though it's less verbose, we have lots of these little edge cases especially around custom components